### PR TITLE
Make supertrait and implied predicates queries defaulted

### PR DESCRIPTION
**Pull request by compiler-errors**
r? @ghost only last commit is meaningful

### CODE REVIEW

**Comment by lqd** [comment]
(and `deep-vector`'s being noisy)

File: compiler/rustc_metadata/src/rmeta/mod.rs [hunk 1]
```diff
@@ -419,10 +421,6 @@ define_tables! {
     lookup_deprecation_entry: Table<DefIndex, LazyValue<attr::Deprecation>>,
     explicit_predicates_of: Table<DefIndex, LazyValue<ty::GenericPredicates<'static>>>,
     generics_of: Table<DefIndex, LazyValue<ty::Generics>>,
-    explicit_super_predicates_of: Table<DefIndex, LazyArray<(ty::Clause<'static>, Span)>>,
-    // As an optimization, we only store this for trait aliases,
-    // since it's identical to explicit_super_predicates_of for traits.
```
Comment (lqd) [inline compiler/rustc_metadata/src/rmeta/mod.rs:424]: drive-by q: was this comment inaccurate maybe?

Comment (compiler-errors) [reply compiler/rustc_metadata/src/rmeta/mod.rs:424]: It was. I changed it when I made associated type bounds be implied in supertrait position. I don't have the PR but you could look.
